daemon/debugtrap.go

```diff
@@ -0,0 +1,38 @@
+package daemon
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/davecgh/go-spew/spew"
+)
+
+// dumpDaemon is a helper function to dump the daemon data-structures
+// for debugging purposes. NO LOCKS ARE TAKEN! (Useful for deadlocks).
+func (d *Daemon) dumpDaemon(root string) {
+	// Ensure we recover from a panic as we are doing this without any locking
+	defer func() {
+		if r := recover(); r != nil {
+		}
+	}()
+
+	if root == "" {
+		logrus.Infof("=== BEGIN daemon dump ===\n%s\n=== END daemon dump ===", spew.Sdump(d))
+	} else {
+		path := filepath.Join(root, "daemon-data.log")
+		f, err := os.OpenFile(path, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0666)
```

Comment:
Carry over from writing the stack trace dump which does the same. It's effectively a no-op on Windows anyway as there's no mode flags in the same way as for Linux.